qdata/parser.py

```diff
@@ -287,16 +385,32 @@ def uop(self, *args):
         return Gate(op_name, params, wires)
 
     def anylist(self, *args):
-        # either <idlist> or <mixedlist>
+        """Either ``idlist`` or ``mixedlist``"""
         return ParsedList(Lists.ANYLIST, unpack(args))
 
     def idlist(self, *args):
+        """Either a single id, or a list of id's. Note that
+        an id is an alphanumeric label (with underscores allowed)
+        that *must* begin with a lowercase character."""
         # <id>
         # <idlist>, <id>
         flat_list = flatten(*args)
         return ParsedList(Lists.IDLIST, flat_list)
 
     def mixedlist(self, *args):
+        """A mixed list is a list that includes a combination
+        (of comma separated) id's, and id's with a non-negative integer index (specified
+        via square brackets).
+
+        .. code-block:: text
+
+            id "[" nninteger "]"
+            mixedlist "," id
+            mixedlist "," id "[" nninteger "]"
+            idlist "," id "[" nninteger "]"
+        """
+        # TODO: This definition doesn't really make sense. The mixedlist
+        # definition could be re-written using `argument`.
```

Comment:
I think `argument` was deliberately kept separate so that the name of the symbol matched the options. That left `mixedlist` to deal with all the other combinations of those things (which maybe didn't suit the name `argument`)?